mfsetup/grid.py

```diff
@@ -728,9 +728,18 @@ def setup_structured_grid(xoff=None, yoff=None, xul=None, yul=None,
     if parent_model is not None and not snap_to_NHG:
 
         # get location of coinciding cell in parent model for upper left
-        pi, pj = parent_model.modelgrid.intersect(xul, yul)
-        verts = np.array(parent_model.modelgrid.get_cell_vertices(pi, pj))
-        xul, yul = verts[:, 0].min(), verts[:, 1].max()
+        # navigating the vertices of the parent cell depends on rotation angle
+        if rotation == 0:
+            pi, pj = parent_model.modelgrid.intersect(xul, yul)
+            verts = np.array(parent_model.modelgrid.get_cell_vertices(pi, pj))
+            xul, yul = verts[:, 0].min(), verts[:, 1].max()
+        else:
+            # if there is rotation, need to make sure we locate within the actual upper left.
+            pi, pj = parent_model.modelgrid.intersect(xul + (delc_grid/100), yul)
+            verts = np.array(parent_model.modelgrid.get_cell_vertices(pi, pj))
+            # upper left vertex has smallest x coordinate under rotation
+            
+            xul, yul = verts[np.argmin(verts[:,0])]
```

Comment:
For clockwise rotation though, won't the smallest x be the *lower left* corner of the parent cell? Is there a way to get the cell vertices in the model coordinate system? That would allow the upper left of the cell to be identified, regardless if the rotation is CW or CCW.

Reply:
there is a `get_local_coords(x,y)` method on the flopy [grid](https://flopy.readthedocs.io/en/3.3.2/source/flopy.discretization.grid.html) object. I guess we could grab all the vertex coords, find the upper left (minx, maxy) and then flip back to world coords using `get_coords(x,y)`. I'll try that out. Perhaps we could also use that to choose `pi` and `pj` above.